core/src/main/java/dev/streamx/cli/exception/LicenseException.java

```diff
@@ -0,0 +1,38 @@
+package dev.streamx.cli.exception;
+
+import java.net.URI;
+
+public class LicenseException extends RuntimeException {
+
+  private LicenseException(String message, Exception exception) {
+    super(message, exception);
+  }
+
+  private LicenseException(String message) {
+    super(message);
+  }
+
+  public static LicenseException licenseFetchException(URI url) {
+    return new LicenseException("""
+        File '%s' couldn't be fetched.
```

Comment:
`File '%s' couldn't be fetched.` this part of exception message seems to be too detailed. I would replace it with something like: `License could not be verified.`

Reply:
Yes, I also was wondering about it. However short and too general message can limit troubleshooting. (I suppose there will be some firewall/proxy/network issues and providing URL may be helpful)

I also asked about this here, but there was no conversation about it https://dynamicsoluti-dlm7051.slack.com/archives/C06MJHTN9PF/p1715854414563169

Reply:
I asked @michalcukierman and we shouldn't place information about URL, so I will delete it